plugins/filter/merge_data.py

```diff
@@ -0,0 +1,335 @@
+# -*- coding: utf-8 -*-
+# Copyright (c) 2022     , DEMAREST Maxime <maxime@indelog.fr>
+# GNU General Public License v3.0+ (see COPYING or https://www.gnu.org/licenses/gpl-3.0.txt)
+
+from __future__ import (absolute_import, division, print_function)
+__metaclass__ = type
+
+DOCUMENTATION = '''
+name: merge_data
+short_description: Merge dict or lists by ensuring the comparing datas are
+                   presents, absents or identics.
+version_added: 5.0.2
+author: DEMAREST Maxime (@indelog)
+description:
+  - Merge two datas, current and expected in form of dict or list.
+  - You can have mixed data like dict->list or list->dict. If the two type
+    are different, the expected datas are always returned.
+  - You have 3 type of merging. You can ensure that the expected data
+    will be present or absent to the result or ensure that the result is
+    identical to the expected datas.
+  - For the list, you have two type of diff. You can check if the expected
+    datas are present in a list on the current datas or you can
+    check a value at the specific index of the list.
+positional: expected, state, list_diff_type
```

Comment:
```suggestion
positional: expected
```
I would keep `state` and `list_diff_type` as keyword arguments.